src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,38 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add movie " + movie + " to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            return Optional.ofNullable(session.get(Movie.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get movie by id " + id + " from DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
```

Comment:
Since you are only reading data from the database, you can use the try-with-resources statement for the Session object here as well. This will ensure that the session is closed automatically.